daemon/src/model/cfd.rs

```diff
@@ -2663,77 +2651,99 @@ mod tests {
     #[test]
     fn ensure_collaborative_settlement_takes_rollover_fees_into_account() {
         let quantity = Usd::new(dec!(10));
-        let leverage = Leverage::new(2).unwrap();
         let opening_price = Price::new(dec!(10000)).unwrap();
         let closing_price = Price::new(dec!(10000)).unwrap();
         let positive_funding_rate = dec!(0.0001);
 
-        assert_eq!(
-            calculate_long_margin(opening_price, quantity, leverage),
-            Amount::from_sat(50000)
-        );
-        assert_eq!(
-            calculate_short_margin(opening_price, quantity),
-            Amount::from_sat(100000)
-        );
+        // Expected payout at closing-price-interval defined by payout curve
+        let expected_payout_taker = Amount::from_sat(49583);
+        let expected_payout_maker = Amount::from_sat(100247);
+        // Expected initial funding fee based on the funding rate and short-margin (because the rate
+        // is positive meaning long pays short)
+        let expected_initial_funding_fee = Amount::from_sat(10);
+
+        // Note that transaction fees ARE zero for these tests because the order's fee-rate is set
+        // to 0!
```

Comment:
Wouldn't it be better to set it so a non-zero value? (Can be done in a follow-up ofc).

Reply:
When starting to get `TxFeeRate` into the tests I noticed that we are actually not taking transaction fees into account for the payout calculation. Should the output of the transaction include this? If I configure the test with a `TxFeeRate` then the amount in the collab settlement tx remains the same.

Reply:
When do the transaction fees come into affect? Would we have to extract the payout differently from the transaction so fees are taken into account?

Reply:
After discussion, here is my understanding:
Collab close tx fee is hardcoded to `1 sat/vbyte` resulting in `85` sats per party to be paid. 
The lock tx fee is irrelevant because it is paid on top of the lock amount when publishing lock.
We construct lock together and when it is build `amount + fees` is constructed, where the `fee` is collaboratively paid by both parties.

I tried to reflect the settlement tx fee with a constant to make this clearer.